### PR TITLE
[snmpagent] Fix hardcoded qsfp lane count by reading sensor status from DB

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- What I did**

The current snmpagent implementation hardcoded qsfp lanes to 4, but qsfp-dd has 8 lanes. In that case snmp query only display the sensors of first 4 lanes. This PR is to fix it. 

**- How I did it**

1. read actual sensor status from redis db, now we only care temperature, voltage, txpower, rxpower, txbias.
2. sort sensor data to make sure it has a solid order
3. store the data to snmpagent data structure

**- How to verify it**

Manual test on master and run existing regression

**- Description for the changelog**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@qiluo-msft can you please review? please note this is for master and should not be cherry pick for 20911. There is a different PR for 201911 as code is different.

File: src/sonic_ax_impl/mibs/ietf/rfc2737.py
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/ietf/rfc2737.py:172]: >if_alias [](start = 20, length = 8)

Why `lane_number == 0` is treated differently? Should it be `<if_alias>/0` ?

File: src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py:21]: >TransceiverSensorData [](start = 6, length = 21)

These classes seem over engineering. You don't use `self` much, so it is belonging to the same type with different fields.

File: src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py:143]: >SORT_FACTOR [](start = 4, length = 11)

Distributed the SORT_FACTOR is not a good ideas. Better to implement a centralized sort key function.

**Review by qiluo-msft — COMMENTED** [review]
As comments

File: src/sonic_ax_impl/mibs/ietf/rfc2737.py
Comment (Junchao-Mellanox) [reply src/sonic_ax_impl/mibs/ietf/rfc2737.py:172]: lane number is 1 based. 0 means the sensor does not belong to any lane but for the cable.

File: src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py
Comment (Junchao-Mellanox) [reply src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py:21]: Hi @qiluo-msft , i am thinking how to change these classes to fields. One way might work could be using a class level dictionary to save different sensor data for different sensor types. Some code could be like:

```python
class TransceiverSensorData:
        sensor_data = {
        'temperature': {
            'name': 'Temperature',
            'oid_offset_base': SENSOR_TYPE_TEMP,
            'sort_factor': 0
        },
        'voltage': {
            'name': 'Voltage',
            'oid_offset_base': SENSOR_TYPE_VOLTAGE
            'sort_factor': 9000
        },
        'rx(\d+)power': {
            'name': 'RX Power',
            'oid_offset_base': SENSOR_TYPE_VOLTAGE,
            'sort_factor': 2000
        }
    }
```

Any suggestion?

**Review by qiluo-msft — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@qiluo-msft should we wait for additional approval or can this be merged and taken to master via submodule PR update?

**Pull request merged**
